### PR TITLE
Added: #1076 Custom naming pem and key files

### LINKED ISSUE #1076: Custom naming pem and key files
**Question**
I was wandering if it would be possible to change the name of the pem and key files while creating them. Something like cosign generate-key-pair --name project-3

### PR DESCRIPTION
**Pull request by Knibbaz**
Closes #1076

You as a user can now give the files a custom name, if you are lazy the files still get the name cosign.pub and cosign.key.

### CODE REVIEW

**Comment by Knibbaz** [comment]
Thought I didn't add the second commit

**Comment by dlorenc** [comment]
Looks like you missed the DCO still - can you get that signed so we can take a look?

File: cmd/cosign/cli/generate/generate_key_pair.go [hunk 1]
```diff
@@ -55,13 +54,17 @@ func GenerateKeyPairCmd(ctx context.Context, kmsVal string, args []string) error
...
-		if err := os.WriteFile("cosign.pub", pemBytes, 0600); err != nil {
+		if err := os.WriteFile(name+".pub", pemBytes, 0600); err != nil {
 			return err
 		}
-		fmt.Fprintln(os.Stderr, "Public key written to cosign.pub")
+		fmt.Fprintln(os.Stderr, "Public key written to "+name+".pub")
 		return nil
 	}
 
+	if name == "" {
```
Comment (developer-guy) [inline cmd/cosign/cli/generate/generate_key_pair.go:64]: why don't you just specify this value as the flag's default value, wouldn't it be better?

```go
cmd.Flags().StringVar(&o.Name, "name", "cosign", "adds a choosen name to your .pub and .key files")
```

File: cmd/cosign/cli/generate/generate_key_pair.go [hunk 2]
```diff
@@ -86,9 +89,9 @@ func GenerateKeyPairCmd(ctx context.Context, kmsVal string, args []string) error
 		return err
 	}
 
-	if fileExists("cosign.key") {
+	if fileExists(name + ".key") {
```
Comment (developer-guy) [inline cmd/cosign/cli/generate/generate_key_pair.go:92]: maybe we can create a local variable that holds this whole name, then we can use it everywhere we need, WDYT?

**Review by developer-guy — CHANGES_REQUESTED** [review]

**Comment by dlorenc** [comment]
Closing, please reopen if you get DCO working.

**Comment by patflynn** [comment]
@Knibbaz Do you think you'll be able  to get DCO working and resubmit?  If not let me know and I'll push it up for you.

**Pull request closed without merging**